kingpin/actors/rightscale/test/integration_mci_and_st.py

```diff
@@ -0,0 +1,138 @@
+                     'images': self.images},
+            dry=True)
+        yield actor.execute()
+
+    @attr('rightscale', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_01b_create_mci(self):
+        actor = mci.MCI(
+            options={'name': self.name,
+                     'description': self.name,
+                     'state': 'present',
+                     'images': self.images})
+        yield actor.execute()
+
+    @attr('rightscale', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_03a_update_mci(self):
+        self.images[0]['instance_type'] = 'm1.large'
+        self.images[1]['instance_type'] = 'm1.large'
+
+        actor = mci.MCI(
+            options={'name': self.name,
+                     'description': self.name,
+                     'state': 'present',
+                     'commit': 'Changed instance type to m1.large',
+                     'images': self.images})
+
+        yield actor.execute()
+
+    @attr('rightscale', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_03b_update_mci(self):
+
+        actor = mci.MCI(
+            options={'name': self.name,
+                     'description': self.name,
+                     'state': 'present',
+                     'tags': 'some_new_tag',
+                     'commit': 'Added a tag',
+                     'images': self.images})
+
+        yield actor.execute()
+
+    @attr('rightscale', 'integration', 'dry')
+    @testing.gen_test(timeout=60)
+    def integration_05a_create_st_dry(self):
+        actor = server_template.ServerTemplate(
+            options=self.template,
+            dry=True)
+        yield actor.execute()
+
+    @attr('rightscale', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_05b_create_st(self):
+        actor = server_template.ServerTemplate(
+            options=self.template)
+        yield actor.execute()
+
+    @attr('rightscale', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_06a_delete_st(self):
+        self.template['state'] = 'absent'
+        actor = server_template.ServerTemplate(
+            options=self.template)
+        yield actor.execute()
+
+    @attr('rightscale', 'integration', 'dry')
+    @testing.gen_test(timeout=60)
+    def integration_09a_destroy_mci_dry(self):
+        actor = mci.MCI(
+            options={'name': self.name,
+                     'description': self.name,
+                     'state': 'absent',
+                     'images': self.images},
+            dry=True)
+        yield actor.execute()
+
+    @attr('rightscale', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_09b_destroy_mci(self):
+        actor = mci.MCI(
+            options={'name': self.name,
+                     'description': self.name,
+                     'state': 'absent',
+                     'images': self.images})
+        yield actor.execute()
```

Comment:
Yeah so the purpose of these tests is to ensure that the API calls function and don't error out in unexpected ways. I agree that we're not testing the final result, but frankly doing that would be a _ton_ of code for minimal value. The benefit of doing this is to ensure that when RightScale changes their API and breaks something, we just see the tests fail because the API broke.